lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -160,68 +160,59 @@
+# The following example demonstrates the use of [FinetuningScheduler](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html#finetuning_scheduler.fts.FinetuningScheduler) to fine-tune a small foundational model on the [RTE](https://huggingface.co/datasets/viewer/?dataset=super_glue&config=rte) task of [SuperGLUE](https://super.gluebenchmark.com/). Iterative early-stopping will be applied according to a user-specified schedule.
 #
 
 # %%
 import os
 import warnings
 from datetime import datetime
-from importlib import import_module
 from typing import Any, Dict, List, Optional
 
-import datasets
+from packaging.version import Version
 
 import sentencepiece as sp  # noqa: F401 # isort: split
+import datasets
 import pytorch_lightning as pl
 import torch
+from datasets import logging as datasets_logging
 from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
 from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
 from pytorch_lightning.utilities import rank_zero_warn
-from pytorch_lightning.utilities.cli import _Registry
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from torch.optim.adamw import AdamW
 from torch.optim.lr_scheduler import CosineAnnealingWarmRestarts
 from torch.utils.data import DataLoader
 from transformers import AutoConfig, AutoModelForSequenceClassification, AutoTokenizer
 from transformers import logging as transformers_logging
 from transformers.tokenization_utils_base import BatchEncoding
 
-# %%
-# a couple helper functions to prepare code to work with a user module registry
-MOCK_REGISTRY = _Registry()
-
-
-def mock_register_module(key: str, require_fqn: bool = False) -> List:
-    if key.lower() == "finetuningscheduler":
-        mod = import_module("finetuning_scheduler")
-        MOCK_REGISTRY.register_classes(mod, pl.callbacks.Callback)
-    else:
-        raise MisconfigurationException(f"user module key '{key}' not found")
-    registered_list = []
-    # make registered class available by unqualified class name by default
-    if not require_fqn:
-        for n, c in MOCK_REGISTRY.items():
-            globals()[f"{n}"] = c
-        registered_list = ", ".join([n for n in MOCK_REGISTRY.names])
-    else:
-        registered_list = ", ".join([c.__module__ + "." + c.__name__ for c in MOCK_REGISTRY.classes])
-    print(f"Imported and registered the following callbacks: {registered_list}")
-
+if Version(torch.__version__) == Version("1.12.0") or torch.__version__.startswith("1.12.0"):
+    # we need to use a patched version of AdamW to fix https://github.com/pytorch/pytorch/issues/80809
+    # and allow examples to succeed with torch 1.12.0 (this torch bug is fixed in 1.12.1)
+    from fts_examples.patched_adamw import AdamW
+else:
+    from torch.optim.adamw import AdamW
```

Comment:
or let's use `SGD`? since it's just an example.

Reply:
If it's okay, I'd prefer to stick with ``AdamW`` despite the bug since I use that optimizer in other examples and it might be nicer to have things as similar as possible.

Reply:
cool!